zk_ee/src/common_structs/history_map/element_with_history.rs

```diff
@@ -18,6 +18,9 @@ pub struct ElementWithHistory<V, A: Allocator + Clone> {
     pub first: HistoryRecordLink<V>,
     /// Current history record
     pub head: HistoryRecordLink<V>,
+    /// Record that has been committed, or initial if not commit has been
+    /// performed.
+    pub committed: HistoryRecordLink<V>,
```

Comment:
Pls, let's add tests for it. I remember some strange issues when I tried to do something similar